aimon/_utils/_utils.py

```diff
@@ -394,3 +402,38 @@ def lru_cache(*, maxsize: int | None = 128) -> Callable[[CallableT], CallableT]:
         maxsize=maxsize,
     )
     return cast(Any, wrapper)  # type: ignore[no-any-return]
+
+
+def llm_reprompting_function(user_query, 
+                             user_instructions, 
+                             llm_response, 
+                             aimon_response, 
+                             detect,                            # detect object configured by the user using AIMon Detect
+                             extract_response_metadata,         # Function to retrieve context from the LLM response
```

Comment:
why not just add the context here as another field v/s this function?

Reply:
The reprompting function internally handles the call to get a new response from the conservative LLM. This function (extract_response_metadata) is decorated with detect to fetch aimon_response on the new LLM response. 

If I understood this comment correctly, we want the get_response to return new response as well as context, and instead of decorating with detect, we want to directly call aimon (with context, query, instructions and new_response) to fetch the aimon_response? But then we would need to extract the context in the get_response function.